Cgen error with match true

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.12 6762c9a, press to see full `v doctor` output</summary>

|V full version      |V 0.4.12 c221b3226b50b40472864ab261e781a44bf6a5b9.6762c9a
|:-------------------|:-------------------
|OS                  |windows, Microsoft Windows 11 Pro 22635 64-bit
|Processor           |8 cpus, 64bit, little endian, 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz  
|Memory              |35.26GB/63.67GB
|                    |
|V executable        |D:\software\v\v.exe
|V last modified time|2025-09-26 19:51:30
|                    |
|V home dir          |OK, value: D:\software\v
|VMODULES            |OK, value: C:\Users\Sammy\.vmodules
|VTMP                |OK, value: C:\Users\Sammy\AppData\Local\Temp\v_0
|Current working dir |OK, value: D:\Code\v\unnamed_compiler\v_bugs\other_cgen_error_match_true
|                    |
|Git version         |git version 2.51.0.windows.1
|V git status        |weekly.2025.16-830-g6762c9a6
|.git/config present |true
|                    |
|cc version          |cc (Rev2, Built by MSYS2 project) 13.2.0
|gcc version         |gcc (Rev2, Built by MSYS2 project) 13.2.0
|clang version       |clang version 21.1.1
|msvc version        |N/A
|tcc version         |tcc version 0.9.27 (x86_64 Windows)
|tcc git status      |thirdparty-windows-amd64 b425ac82
|emcc version        |emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 4.0.3 (a9651ff57165f5710bb09a5fe52590fd6ddb72df)
|glibc version       |N/A

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  main.v && D:\Code\v\unnamed_compiler\v_bugs\other_cgen_error_match_true\main.exe`
```v
module main

type SecondTokenizerValue = []rune | Keyword

enum AdvancedTokenType {
	identifier
	keyword
	newline
}

struct SecondTokenizerToken {
	type  AdvancedTokenType
	value ?SecondTokenizerValue
}

enum Keyword {
	module
	import
}

fn main() {
	mut tokens := []SecondTokenizerToken{}
	tokens << SecondTokenizerToken{
		type:  AdvancedTokenType.keyword
		value: Keyword.module
	}
	build_ast(tokens) or { assert err.msg() == 'Handling module' }
}

struct RootAST {
mut:
	module []rune
}

fn build_ast(tokens []SecondTokenizerToken) !RootAST {
	for _ in 0 .. 1 {
		token := tokens[0]
		match true {
			false {
				return error('Expected `module` keyword at the start of the file, but got `${token.type}`')
			}
			token.type == .keyword && token.value == ?SecondTokenizerValue(Keyword.module) {
				return error('Handling module')
			}
			else {
				continue
			}
		}
	}

	return error('Not implemented')
}

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: d:/software/v/thirdparty/tcc/include/winapi/synchapi.h:167: warning: CreateMutex redefined
cc: C:/Users/Sammy/AppData/Local/Temp/v_0/main.01K63TCCPD8XTVFB952WKGJ22K.tmp.c:5547: error: '_t4' undeclared
...
cc: C:/Users/Sammy/AppData/Local/Temp/v_0/main.01K63TCCPD8XTVFB952WKGJ22K.tmp.c:5547: error: '_t4' undeclared
(note: the original output was 8 lines long; it was truncated to its first 2 lines + the last line)
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

No cgen error


Related to #25342 and #25372
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/match.v
+++ b/vlib/v/gen/c/match.v
@@ -435,6 +435,11 @@ fn (mut g Gen) match_expr_classic(node ast.MatchExpr, is_expr bool, cond_var str
 	mut has_goto := false
 	for j, branch in node.branches {
 		is_last := j == node.branches.len - 1
+		reset_if = branch.exprs.any(g.match_must_reset_if(it))
+		if reset_if {
+			g.writeln('')
+			g.set_current_pos_as_last_stmt_pos()
+		}
 		if branch.is_else || (use_ternary && is_last) {
 			if node.branches.len > 1 {
 				if use_ternary {
@@ -469,7 +474,6 @@ fn (mut g Gen) match_expr_classic(node ast.MatchExpr, is_expr bool, cond_var str
 				g.write_v_source_line_info(branch)
 				g.write('if (')
 			}
-			reset_if = branch.exprs.any(g.match_must_reset_if(it))
 			for i, expr in branch.exprs {
 				if i > 0 {
 					g.write(' || ')

--- a/vlib/v/tests/options/option_case_test.v
+++ b/vlib/v/tests/options/option_case_test.v
@@ -1,0 +1,52 @@
+module main
+
+type SecondTokenizerValue = []rune | Keyword
+
+enum AdvancedTokenType {
+	identifier
+	keyword
+	newline
+}
+
+struct SecondTokenizerToken {
+	type  AdvancedTokenType
+	value ?SecondTokenizerValue
+}
+
+enum Keyword {
+	module
+	import
+}
+
+fn test_main() {
+	mut tokens := []SecondTokenizerToken{}
+	tokens << SecondTokenizerToken{
+		type:  AdvancedTokenType.keyword
+		value: Keyword.module
+	}
+	build_ast(tokens) or { assert err.msg() == 'Handling module' }
+}
+
+struct RootAST {
+mut:
+	module []rune
+}
+
+fn build_ast(tokens []SecondTokenizerToken) !RootAST {
+	for _ in 0 .. 1 {
+		token := tokens[0]
+		match true {
+			false {
+				return error('Expected `module` keyword at the start of the file, but got `${token.type}`')
+			}
+			token.type == .keyword && token.value == ?SecondTokenizerValue(Keyword.module) {
+				return error('Handling module')
+			}
+			else {
+				continue
+			}
+		}
+	}
+
+	return error('Not implemented')
+}

